### ISSUE TITLE
Handling of always_relations in Chinese

### ISSUE THREAD

**Issue by paynesa** [isi-vista/adam#811]
In English, if we say "I push the book on the table," this can either mean "I push the book that is on the table" or "On the table, I push the book" i.e. we can use the same syntax to represent a PP that modifiers the theme or the path. In both cases, there is an `always_relation` indicating that the book is on the table, and this works in English since our generator technically generates the first meaning when we examine the tree, but the linearized result also gives the second meaning. 
However, in Chinese, the situation is a bit different as these two meaning are syntactically distinct: we can either say:
1. "I (on the table) push the book" -> The entire book pushing action is happening on the table; presumably I am in it too 
2. "I (the book) push on the table" -> The direction of the pushing is along the table (but I could be standing on the floor)
3. "I push (the book on the table)" -> The book is on the table, but I could be about to push it off of the table

We need unique ways to generate each of these syntaxes in order to represent all possible meanings. Currently, we handle this in the Chinese language generator with the following:
* `always_relations` generate the first case since the entire action occurs on the table here, so the book is always on the table.
* `during` relations and `after_action_relations` generate the second type since they indicate path
* `always_relations` can indicate NP modification like in the 3rd case, but currently this is only handled for phrases without actions since we need a way to distinguish between a modification that modifies the NP and the VP, and this is currently distinguished using presence or absence of `action`, meaning we can't currently generate such a syntax as the third one. Notably, I don't think we currently explicitly handle NP modification within a VP in English, in which case this could be irrelevant, but it may also be something we want to handle in the future.

Some potential solutions:

1. Handle `always_relations` as NP relations always and then use `during.continuously` relations for the first type of situation.
2. Add some sort of syntax hint that tells us which way to handle it

**Comment by LichMaster98** [comment]
@paynesa For case 1 could you check if the `always_relations` contains a `contacts` relationship for both the speaker and the book between the table? `contacts([SPEAKER, BOOK], GROUND)` 

I agree with the use of the during relations to generate case two.

For case three I'm unsure where we'd have such a situation currently because what is about to happen after our ending frame isn't observed in any way. If we wanted to assert "I pushed the book off the table" as a situation that would probably be the language equivalent rather than describing that the book was on the table before it got pushed off. With that I'm not sure handling case 3 is as important as cases 1 and 2 for now.

**Comment by gabbard** [comment]
@paynesa : I agree with @LichMaster98  - our two-frame representaiton of actions can represent "pushing the book off the table", but not pushing the book along a trajectory which would soon end up with it falling off the table.

**Comment by paynesa** [comment]
@gabbard @LichMaster98 makes sense!
I think for Jacob's point on case 1, it could get tricky for non-path cases such as `roll,` where we could use case 1 syntax for, say, "I roll the cookie on the table" since the cookie is always on the table and there's no clear path, but that doesn't necessarily mean that I'm also on the table for non-path verbs (i.e. for verbs that don't indicate path, case 2 is equivalent to case 1 so we can only indicate whether the rolling or the cookie is on the table, but we don't know anything about where I am). If we do implement case 3, I could imagine some sort of related hack where we just assume everything's on the table in order to produce case 1 over case 3, if that makes sense.

**Comment by gabbard** [comment]
@paynesa : I wonder if using "the ground" is a better example than "a table".  For example, it's weird for an adult human to do anything with an object on a table while themselves being on the table, but very plausible if they are rolling something "on the ground."  To distinguish those cases you'd need to examine what relations hold for the "roller" as well as the "rollee".

**Comment by gabbard** [comment]
Is case 3 "the book is initially on the table, but there is no guarantee it remains there throughout the action"?

**Comment by paynesa** [comment]
@gabbard That makes sense to me. Is this something I should try to implement or do we want to wait since we're not really dealing with case 3? 

Yes, case 3 would probably be used if there were multiple books and you're distinguishing between them, i.e. "I'm pushing this book that is on the table rather than the one on the floor etc." but path would be indicated separately so you could absolutely say something like "I'm going to push the book that's on the table off of the table" and syntactically you'd combine cases 2 and 3:

"I (the book on the table) push (off the table)"
